File: 58.length-of-last-word.py
#
# @lc app=leetcode id=58 lang=python
#
# [58] Length of Last Word
#

# @lc code=start
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        start = False
        count = 0
        for i in s[::-1]:
            if i !=' ':
                start = True 
            if i == ' ' and start:
                break
            if i != ' ':
                count += 1
        return count

        
# @lc code=end

File: 58.length-of-last-word.java
/*
 * @lc app=leetcode id=58 lang=java
 *
 * [58] Length of Last Word
 */

// @lc code=start
class Solution {
    public int lengthOfLastWord(String s) {
        boolean start = false;
        int count = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) != ' ') {
                start = true;
            }
            if (s.charAt(i) == ' ' && start) {
                break;
            }
            if (s.charAt(i) != ' ') {
                count++;
            }
        }
        return count;
    }
}
// @lc code=end

